iceoryx_posh/source/roudi/port_manager.cpp

```diff
@@ -659,19 +659,43 @@ popo::InterfacePortData* PortManager::acquireInterfacePortData(capro::Interfaces
     }
 }
 
+void PortManager::publishCurrentServiceRegistry() const noexcept
+{
+    // Send the new serviceRegistry here and ring the bell to inform all ServiceDiscovery instances about the change
+    if (m_serviceRegistryPublisherPortData.has_value())
+    {
+        PublisherPortUserType publisher(m_serviceRegistryPublisherPortData.value());
+        auto maybeChunkHeader = publisher.tryAllocateChunk(sizeof(ServiceRegistry),
+                                                           alignof(ServiceRegistry),
+                                                           CHUNK_NO_USER_HEADER_SIZE,
+                                                           CHUNK_NO_USER_HEADER_ALIGNMENT);
```

Comment:
Why not use `and_then` then we do not need a `maybeChunkHeader`?

Reply:
True, no `return` is needed here.